### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by amugofjava** [amugofjava/anytime_podcast_player#72] [Ideas]
The below is a roadmap for next development - subject to change. Any feedback greatly appreciated:

**- Queue support**
Add the ability to queue episodes and manage the queue. _Completed & Released._

**- Transcript support.**
Add support for the Podcasting 2.0 transcript tag. The first phase will allow users to view transcripts where a podcast supports it. The second phase will allow the transcript to be followed as the episode is played - closed captions.  _Completed & Released._

**- Null Safety & Dart 3**
Null safety is to become the default in the near future. Before Anytime can grow it needs to be migrated to null safe code. _Completed & Released._

**- Accessibility enhancements**
Improve accessibility support, especially with TalkBack/Voiceover.

**- Background fetch**
Full fetching and updating of episodes in the background.

**- Sleep timer**
Useful addition with queue completed.

**- Improve discovery**
Add category selection to the search panel using categories provided by the currently selected search provider.

**- Episode view**
Display episodes across all podcasts and allow filtering. For example, show started but not finished episodes.

**Comment by Chralu** [comment]
Hi @amugofjava ,

That's a satisfying roadmap to me !

I would add production deployment of iOS app. 
Indeed, for the moment, iOS version is released on testflight. Unfortunately, Testflight releases are disabled after 90 days.

What do you think ?

**Comment by amugofjava** [comment]
Hi @Chralu,

Thank you.

Yes, the Testflight version has long expired but it was published to production. If it's not on the App Store I have done something very wrong :/ Are you unable to find Anytime on the App Store (other than Testflight)?

https://apps.apple.com/us/app/anytime-podcast-player/id1582300839

Thanks.

**Reply by Chralu** [reply]
I didn't know you had pushed a production release 😅. That's great news, I can switch back to Anytime 🎉 

Your roadmap seems fine to me then 😁
Thank you for the good work !

**Comment by amugofjava** [comment]
I have pushed a new [up-next](https://github.com/amugofjava/anytime_podcast_player/tree/feature/up-next) branch containing the queue feature, and I have pushed an Alpha build to the Google Play Store. Any feedback on the queue support would be greatly appreciated.

If you would like access to the Alpha track to help test that would be fantastic. Drop me a message and I'll get you setup.

**Reply by Chralu** [reply]
Hi @amugofjava ,

I would enjoy to test a Testflight release :)

**Reply by amugofjava** [reply]
Hi @Chralu,

Google Play supports several test tracks which is great for internal, Alpha & Beta tests. Testflight seems to be aimed at Beta tests only. I will be moving to Beta testing soon, so I will definitely be releasing to Testflight then.

**Reply by Chralu** [reply]
Ok I will test it on Android then :)

To use Testflight for internal testing, testers should be added to the developer account. Minimal `sales` or `developer` permissions should be enough.
But I must admit this is a bit intrusive...

**Reply by amugofjava** [reply]
I will do some research on internal testers. Thanks @Chralu.

**Comment by amugofjava** [comment]
I have pushed a new release and released to Beta on both Google Play & TestFlight.

TestFlight: https://testflight.apple.com/join/C9D6ODqx

**Comment by amugofjava** [comment]
I have prioritised migrating the code to null safe after the current transcript work.

**Comment by amugofjava** [comment]
I have inserted a new task into the roadmap to improve accessibility. After a recent review of Anytime's accessibility support I found it to be failing badly with Talkback/Voiceover. Therefore, I feel it is important to improve support for screen readers.

**Comment by amugofjava** [comment]
With 2023 drawing to a close, I thought it a good time to update the state of the road map. 

- Queue support: **Complete**
- Transcript support: **Complete**
- Null Safety & Dart 3: **Complete**
- Accessibility enhancements: **Greatly improved**. Always room for more improvements.
- Sleep timer: **Complete**
- Improve discovery (category selection): **Complete**
- Background fetch: **Todo** - requires DB rewrite before this can be implemented.
- Episode view: **Todo**

**Reply by Chralu** [reply]
Hi @amugofjava ,

About the isolate proof SGBD, have you looked at [Drift](https://drift.simonbinder.eu) ? 
It seems to handle [background worker problematics](https://drift.simonbinder.eu/isolates/).

**Comment by amugofjava** [comment]
Hi @Chralu,

Yea, I have considered a lot of options and have looked at Drift. I've also considered using the underlying sqlite3 library directly as the Anytime schema is fairly simple and it may allows me to use my current models rather than generating new ones.

**Comment by amugofjava** [comment]
Closing this discussion now as only the background fetch item is left and is currently being worked on. Will open a new Roadmap for the rest of 2025-2026 soon.
